### PR TITLE
Fixed locale problems with cgltf.h

### PR DESCRIPTION
**Pull request by ksuprynowicz**
Thanks to @GooseVali for helping with this!
Fixes https://github.com/overte-org/overte/issues/974

### CODE REVIEW

**Comment by JulianGro** [comment]
I tested the AppImage (https://public.overte.org/build/overte/PR978/Overte-2024.05.02-locale-7-g4b9ddbc260-debug-x86_64.AppImage) and it indeed works fine now.

**Comment by ksuprynowicz** [comment]
This just needs code review now and then we can merge it :)

File: libraries/model-serializers/src/GLTFSerializer.h [hunk 1]
```diff
@@ -14,13 +14,29 @@
...
+    streamToParse.imbue(std::locale("C"));
+    float value;
+    if(!(streamToParse >> value)) {
+        qDebug(modelformat) << "cgltf: Cannot parse float from string: " << str;
+        return 0.0f;
+    }
+    return value;
+}
+
+#define CGLTF_ATOF(str) atof_locale_independent(str)
```
Comment (HifiExperiments) [inline libraries/model-serializers/src/GLTFSerializer.h:36]: cool!  how does this work, this macro is called by cgltf if it exists?

**Review by HifiExperiments — APPROVED** [review]

File: libraries/model-serializers/src/GLTFSerializer.h (hunk 1 above)
Comment (ksuprynowicz) [reply libraries/model-serializers/src/GLTFSerializer.h:36]: Yes, here's the relevant part:
```C++
#ifndef CGLTF_ATOF
#define CGLTF_ATOF(str) atof(str)
#endif
```

**Pull request merged**